gadget-container/gadgets/networkpolicyadvisor/main.go

```diff
@@ -158,18 +154,61 @@ func main() {
 	}
 
 	// Start the BPF tracer
-	t, err := tracer.NewTracer(&tcpEventTracer{clientset})
+	mytracer := &tcpEventTracer{
+		clientset: clientset,
+		queue:     make(chan tracer.TcpV4, 500),
+	}
+	t, err := tracer.NewTracer(mytracer)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "%v\n", err)
 		os.Exit(1)
 	}
 
+	ticker := time.NewTicker(time.Second)
```

Comment:
Out of curiosity: Why 1 second?, is that a big enough to about so many calls and small enough to avoid pods terminating before collecting the data?

Reply:
> is that a big enough to about so many calls

In practice it is working fine for me on small clusters. But with many nodes, that could be a problem when each node calls the apiserver every second. It reminds me of https://cilium.io/blog/2019/04/24/cilium-15/ ("Scaling Kubernetes watchers"). 

> small enough to avoid pods terminating before collecting the data?

For the application used in [demo-network-policy.md](https://github.com/kinvolk/inspektor-gadget/blob/master/Documentation/demo-network-policy.md), it's fine because the use case is to start the deployments and let them run for a while. But for short-lived pods like Jobs that might not work, we might need a cache with a PodInformer...

Reply:
We can just open an issue to keep track of this limitation. It seems to be a complex problem to me.